hack/validate-vendor.bash

```diff
@@ -8,26 +8,32 @@
 
 set -e -o pipefail
 
-# TRAVIS_BRANCH is the name of the branch targeted by the pull request.
-# but we won't have that set if running locally
-if [ -z "$TRAVIS_BRANCH" ]; then
-	VALIDATE_REPO='git@github.com:golang/dep.git'
+if [ -z "$VALIDATE_UPSTREAM" ]; then
+	VALIDATE_REPO='https://github.com/golang/dep.git'
 	VALIDATE_BRANCH='master'
 
+	VALIDATE_HEAD="$(git rev-parse --verify HEAD)"
+
 	git fetch -q "$VALIDATE_REPO" "refs/heads/$VALIDATE_BRANCH"
-	TRAVIS_BRANCH="$(git rev-parse --verify FETCH_HEAD)"
-fi
+	VALIDATE_UPSTREAM="$(git rev-parse --verify FETCH_HEAD)"
 
-VALIDATE_HEAD="$(git rev-parse --verify HEAD)"
+	VALIDATE_COMMIT_DIFF="$VALIDATE_UPSTREAM...$VALIDATE_HEAD"
+
+	validate_diff() {
+		if [ "$VALIDATE_UPSTREAM" != "$VALIDATE_HEAD" ]; then
+			git diff "$VALIDATE_COMMIT_DIFF" "$@"
+		fi
+	}
+fi
 
 IFS=$'\n'
-files=( $(git diff "$TRAVIS_BRANCH...$VALIDATE_HEAD" --diff-filter=ACMR --name-only -- 'manifest.json' 'lock.json' 'vendor/' || true) )
+files=( $(validate_diff --diff-filter=ACMR --name-only -- 'manifest.json' 'lock.json' 'vendor/' || true) )
 unset IFS
 
 if [ ${#files[@]} -gt 0 ]; then
 	# We run ensure to and see if we have a diff afterwards
-	go build
-	./dep ensure
+	go install ./cmd/dep
+	dep ensure
```

Comment:
Depending on how PATH is setup, this isn't guaranteed to use the binary we just built. Can we fix the build command and continue using `./dep`?